klippy/extras/display/menu.py

```diff
@@ -33,21 +33,33 @@ def get_prefix_options(self, prefix):
 
 # Scriptable menu element abstract baseclass
 class MenuElement(object):
-    def __init__(self, manager, config):
+    def __init__(self, manager, config, default_name=None):
         if type(self) is MenuElement:
             raise error(
                 'Abstract MenuElement cannot be instantiated directly')
         self._manager = manager
         self.cursor = '>'
-        # scroll is always on
-        self._scroll = True
-        self._index = manager.asint(config.get('index', ''), None)
-        self._enable_tpl = manager.gcode_macro.load_template(
-            config, 'enable', 'True')
-        self._name_tpl = manager.gcode_macro.load_template(
-            config, 'name')
         # item namespace - used in relative paths
         self._ns = str(" ".join(config.get_name().split(' ')[1:])).strip()
+        # scroller is always on
+        self._scroll = True
+        self._index = manager.asint(config.get('index', ''), None)
+        # test enable for static value
+        # in case of static boolean value, don't use template
+        _enable_value = config.get('enable', 'True')
+        self._enable_sta = manager.asbool(_enable_value, None)
+        self._enable_tpl = None
+        if self._enable_sta is None:
+            # non static boolean value, use template
+            self._enable_tpl = manager.gcode_macro.create_template(
+                '%s:enable' % (self._ns, ), _enable_value)
```

Comment:
Use:
```python
    self._enable_tpl = self._enable_sta = None
    try:
        self._enable_sta = config.getboolean('enable', True)
    except config.error as e:
        self._enable_tpl = manager.gcode_macro.load_template(config, 'enable')
```

The above doesn't require changes to gcode_macro.py .

Reply:
Thanks. I'll try to redo it.